torchmetrics/timeseries.py

```diff
@@ -0,0 +1,48 @@
+from typing import List, Tuple, Union
+
+import torch
+from torch import nn
+
+from torchmetrics.metric import Metric
+
+
+class TimeSeriesMetric(nn.ModuleList):
+    """
+    A class that keeps track of metrics as a timeseries, and implements useful methods.
+
+    Args:
+        metric: the torchmetric to keep track of at each timestep.
+    """
+
+    def __init__(self, metric: Metric):
+        super().__init__()
+        self.metric = metric
+
+    @property
+    def n_timesteps(self) -> int:
```

Comment:
```suggestion
    def num_timesteps(self) -> int:
```
or `nb_timesteps` but is it needed if it just len? or later it can be different?

Reply:
well, shall it rather count of all preds/targets?